app/javascript/components/applications/application_form/EligibilitySection.js

```diff
@@ -1,161 +1,112 @@
-          blockNote='(required)'
-          fieldName='is_first_time_homebuyer'
-          ariaLabelledby='prereqs'
-          validation={validate.isChecked(this.checkboxErrorMessage)}
-        />
-      </div>
-    )
-  }
 
-  completedHomebuyersEducationMarkup = () => {
-    return (
-      <div className='small-12 columns'>
-        <CheckboxField
-          id='has_completed_homebuyer_education'
-          label={`Completed homebuyers' education`}
-          blockNote='(required)'
-          fieldName='has_completed_homebuyer_education'
-          ariaLabelledby='prereqs'
-          validation={validate.isChecked(this.checkboxErrorMessage)}
-        />
-      </div>
-    )
-  }
-
-  loanPreapprovalMarkup = () => {
-    return (
-      <div className='small-12 columns margin-bottom'>
-        <CheckboxField
-          id='has_loan_preapproval'
-          label='A loan pre-approval letter from a MOHCD-approved lender'
-          blockNote='(required)'
-          fieldName='has_loan_preapproval'
-          ariaLabelledby='prereqs'
-          validation={validate.isChecked(this.checkboxErrorMessage)}
-        />
-      </div>
-    )
+    setLenders(mappedLenders)
   }
+  // With the empty array passed as the second argument
+  // this useEffect call acts like a componentDidMount call
+  useEffect(() => {
+    loadOnMount()
+  }, [])
 
-  handleSelectInstitution = (event) => {
-    const { lendingInstitutions, form } = this.props
-    const lenders = map(lendingInstitutions[event.target.value], (lender) => ({label: `${lender.FirstName} ${lender.LastName}`, value: lender.Id}))
+  const handleSelectInstitution = (event) => {
+    const mappedLenders = map(lendingInstitutions[event.target.value], (lender) => ({label: `${lender.FirstName} ${lender.LastName}`, value: lender.Id}))
     // reset lending_agent select
     form.change('lending_agent', null)
-    this.setState({lenders: lenders})
-  }
-
-  lendingInstitutionMarkup = (lendingInstitutions) => {
-    const institutionOptions = map(lendingInstitutions, (_, key) => ({label: key, value: key}))
-    return (
-      <div className='small-6 columns'>
-        <SelectField
-          label='Name of Lending Institution'
-          blockNote='(required)'
-          id='lending_institution'
-          fieldName='lending_institution'
-          options={institutionOptions}
-          onChange={this.handleSelectInstitution}
-          validation={validate.isPresent('Please select a lending institution.')}
-        />
-      </div>
-    )
-  }
-
-  lenderMarkup = (lenders) => {
-    return (
-      <div className='small-6 columns margin-bottom--2x'>
-        <SelectField
-          label='Name of Lender'
-          blockNote='(required)'
-          id='lending_agent'
-          fieldName='lending_agent'
-          options={lenders}
-          validation={validate.isPresent('Please select a lender.')}
-        />
-      </div>
-    )
-  }
-
-  checkboxesMarkup = () => {
```

Comment:
curious to hear what the motivation is around switching from this "markup" strategy to putting all the html in the return directly.

Reply:
This structure of calling multiple class functions directly like this is another performance issue. The other option would be to rip these out of the class and create small functional components and pass them as a component instead of a function. That way they are handled by react and the virtual dom in a more efficient way and can be loaded ahead of time instead of having these non component functions being re-called on each render

Reply:
ok, this change makes sense. thanks for the explanation!